src/components/ImageUpload.tsx

```diff
@@ -1,66 +1,130 @@
-import { Paper, Typography } from "@mui/material";
-import { styled } from "@mui/material/styles";
-import { Box } from "@mui/system";
-import { Field, FormikErrors, FormikTouched } from "formik";
-import React from "react";
-import { render } from "react-dom";
+import { Box, CircularProgress, Typography } from "@mui/material";
+import { Image } from "cloudinary-react";
+import { useCallback, useState } from "react";
 import { useDropzone } from "react-dropzone";
-import { theme } from "./../theme/theme";
-import { StyledPaperInput } from "./StyledComponents";
+import { StyledFilledButton, StyledPaperInput } from "./StyledComponents";
+
+// once we have backend authentication we will need to protect these variables
+const PUBLIC_CLOUDINARY_CLOUD_NAME = "projectnewm",
+  PUBLIC_CLOUDINARY_UPLOAD_PRESET = "rvktckuk";
 
 interface ImageUploadProps {
-  setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
-  errors: FormikErrors<{
-    description: string;
-    genre: string;
-    imageFile: {
-      name: string;
-      size: string;
-      type: string;
-    };
-  }>;
-  touched: FormikTouched<{
-    description: string;
-    genre: string;
-    imageFile: {
-      name: string;
-      size: null;
-      type: string;
-    };
-    releaseDate: string;
-    title: string;
-    yourRole: string;
-  }>;
+  fileType?: "png" | "jpg" | "jpeg" | "wdp" | "jp2" | "bmp" | "pdf" | "tiff" | "ico" | "eps";
+  dropzoneLabel?: string;
+  buttonLabel?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setFieldValue?: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
+  setTouched?: (fields: { [field: string]: boolean }, shouldValidate?: boolean) => void;
 }
 
-export const ImageUpload = (props: ImageUploadProps) => {
-  const { setFieldValue } = props;
+export const ImageUpload = ({
+  fileType = "png",
+  dropzoneLabel = "Drag & Drop (Square Image Only)",
```

Comment:
Out of curiosity, why square images only?

Reply:
If you submit a non-square image it gets stretched. Another option could be to crop the image to a square, but that could cut out parts that the user may want.  I thought the best option would be to just ensure the user submits a square image. Album covers are square already, so is this copy even necessary?